pkg/gameservers/controller.go

```diff
@@ -416,6 +422,33 @@ func (c *Controller) syncGameServerCreatingState(gs *v1alpha1.GameServer) (*v1al
 	return gs, nil
 }
 
+// syncGameServerCreatingState checks if the GameServer is in the Creating state, and if so
+// creates a Pod for the GameServer and moves the state to Starting
+func (c *Controller) syncDevelopmentGameServer(gs *v1alpha1.GameServer) (*v1alpha1.GameServer, error) {
+	// Get the development IP address
+	devIPAddress, isDevGs := gs.GetDevAddress()
+	if !isDevGs {
+		return gs, nil
+	}
+
+	c.logger.WithField("gs", gs).Infof("%s is a development game server and will not be managed by Agones.", gs.Name)
+
+	gsCopy := gs.DeepCopy()
+	ports := []v1alpha1.GameServerStatusPort{}
+	for _, p := range gs.Spec.Ports {
+		ports = append(ports, p.Status())
+	}
+	gsCopy.Status.State = v1alpha1.GameServerStateReady
+	gsCopy.Status.Ports = ports
+	gsCopy.Status.Address = devIPAddress
+	gsCopy.Status.NodeName = devIPAddress
+	gs, err := c.gameServerGetter.GameServers(gs.ObjectMeta.Namespace).Update(gsCopy)
```

Comment:
Shall we take our new, fancy `UpdateStatus` for a spin here instead?
https://github.com/GoogleCloudPlatform/agones/blob/master/pkg/client/clientset/versioned/typed/stable/v1alpha1/gameserver.go#L40

We'll want to move everything over eventually, so might as well do new work with it.

(Pretty sure you're only updating status here?)

Reply:
@aLekSer - sounds like we can't actually do this one? So I may have to rescind this suggestion.

We would have to wait for your status subresource updates?

Reply:
I'll add a TODO for this.

Reply:
Yes, I think this change could be performed as a separate PR. Cause we need to change Helm template and this could lead to changes in other places too (RBAC permissions for "gameservers/status").
https://github.com/kubernetes/community/blob/master/contributors/design-proposals/api-machinery/customresources-subresources.md